spicy/toolchain/src/compiler/parser/parser.yy

```diff
@@ -781,10 +781,14 @@ unit_property : PROPERTY opt_attributes ';'      { $$ = builder->typeUnitItemPro
               | PROPERTY '=' base_type_no_ref ';'       { $$ = builder->typeUnitItemProperty(ID(std::move($1)), builder->expressionType(builder->qualifiedType(std::move($3), hilti::Constness::Mutable)), {}, false, __loc__); };
 
 
-unit_field    : opt_unit_field_id ':' opt_skip base_type opt_unit_field_repeat opt_attributes opt_unit_field_condition opt_unit_field_sinks opt_unit_item_hooks
```

Comment:
This is nice and compact, but misses using the same item format for the following input :/

```
type MyBytes = bytes;

public type BytesVectors = unit {
    : MyBytes[10] &size=5;
};
```

Reply:
This case too 🤦 

```
public type BytesVectors = unit {
    : (bytes[4] &size=6)[10] &size=5;
};
```

I think this PR will remain just fixing the issue then #1855 will be to try to clean up the grammar so that all of these could be more intuitive to catch, I think there's potential to make it a fair bit nicer but maybe not in this PR since I'd want to dig in a fair bit deeper

Reply:
The issue I mentioned offline was actually with

```ruby
type MyBytes = bytes;
```

This is accepted with the current patch, even though it would cause the issue described in #1832 (albeit through an alias). I believe this PR should address this so we can close the issue.

---

You are seeing an error about "missing `&size`" with

```ruby
type MyBytes = bytes &size=4;
```

since the `&size` is a parsing attribute and it is ignored in the declaration of the type alias. This came up before and I just filed https://github.com/zeek/spicy/issues/1861 to address this by adding a diagnostic.

Reply:
OK so for the case I pointed out I just went in and modified the test `offset-parse-from-vector.spicy` to not use attributes in a type alias so it doesn't come up